src/test/components/TimelineFlameChart.test.js

```diff
@@ -0,0 +1,66 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// @flow
+import React from 'react';
+import TimelineFlameChart from '../../components/timeline/TimelineFlameChart';
+import renderer from 'react-test-renderer';
+import { Provider } from 'react-redux';
+import mockCanvasContext from '../fixtures/mocks/canvas-context';
+import { storeWithProfile } from '../fixtures/stores';
+import { getBoundingBox } from '../fixtures/utils';
+import { getProfileForUnfilteredCallTree } from '../fixtures/profiles/profiles-for-call-trees';
+import { changeTimelineFlameChartExpandedThread } from '../../actions/timeline';
+
+jest.useFakeTimers();
+
+it('renders TimelineFlameChart correctly', () => {
+  // Tie the requestAnimationFrame into jest's fake timers.
+  window.requestAnimationFrame = fn => setTimeout(fn, 0);
+  window.devicePixelRatio = 1;
+  const ctx = mockCanvasContext();
+
+  /**
+   * Mock out any created refs for the components with relevant information.
+   */
+  function createNodeMock(element) {
+    // <TimelineCanvas><canvas /></TimelineCanvas>
+    if (element.type === 'canvas') {
+      return {
+        getBoundingClientRect: () => getBoundingBox(200, 300),
+        getContext: () => ctx,
+        style: {},
+      };
+    }
+    // <TimelineViewport />
+    if (element.props.className.split(' ').includes('timelineViewport')) {
+      return {
+        getBoundingClientRect: () => getBoundingBox(200, 300),
+      };
+    }
+    return null;
+  }
+
+  const store = storeWithProfile(getProfileForUnfilteredCallTree());
```

Comment:
I don't know which profile is best to use. This one produced some results in the snapshot, but I guess this is not a call tree test.

Reply:
That one will work great. All I meant by the function name was that it makes a nice a call tree.